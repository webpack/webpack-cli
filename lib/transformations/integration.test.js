const fs = require('fs');
const path = require('path');
const transform = require('./').transform;
const no_of_files = 1;

const runTest = (currentConfigPath) => {

	let currentConfig = fs.readFileSync(currentConfigPath, 'utf8');
	const outputConfig = transform(currentConfig);
	expect(outputConfig).toMatchSnapshot();
};

describe('Run integration tests', () => {
	for(let i =1; i<= no_of_files; i++){
		test(`run test - webpack${i}.config.js`, () => {
			const filePath = path.resolve(process.cwd(), './lib/transformations/__testfixtures__/webpack' + i + '.config.js');
			runTest(filePath);
		});
	}
});
