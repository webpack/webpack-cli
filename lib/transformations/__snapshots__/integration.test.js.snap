// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Run integration tests run test - webpack1.config.js 1`] = `
"const ExtractTextPlugin = require('extract-text-webpack-plugin');
const webpack = require('webpack');
const srcPath = '../dummy';
const distPath = '../dummy/dist';

module.exports = {
	context: srcPath,
	entry: {
		foo: './css/pages/foo.css',
		bar: './css/pages/bar.css'
	},
	output: {
		path: distPath,
		publicPath: '/assets/',
		filename: '[name].js'
	},
	module: {
		rules: [{
			test: /\\\\.css$/,
			use: ExtractTextPlugin.extract([
				'css-loader'
			])
		}]
	},
	plugins: [
		new webpack.optimize.CommonsChunkPlugin({
			name: 'common',
			minChunks: 2
		}),
		new ExtractTextPlugin({
			filename: 'css/[name].[contenthash:base64:5].css',
			allChunks: true
		})
	]
};
"
`;
